vector/src/main/java/im/vector/app/core/utils/PermissionsTools.kt

```diff
@@ -137,6 +138,32 @@ fun checkPermissions(
     }
 }
 
+/**
+ * Checks if required WebView permissions are already granted system level.
+ * @param activity the calling Activity that is requesting the permissions (or fragment parent)
+ * @param request WebView permission request of onPermissionRequest function
+ * @return true if WebView permissions are already granted, false otherwise
+ */
+fun checkWebViewPermissions(activity: Activity, request: PermissionRequest): Boolean {
```

Comment:
Maybe we could extract this method into a dedicated use case such as `CheckWebViewPermissionsUseCase`? We could also add unit tests on it as bonus.

Reply:
Good idea, done.